### PR TITLE
[UBS-User. User Data page] User Date don't save changes if house number with "/" like "1/3"

### PR DESCRIPTION
**Pull request by ABbondar**
# GreenCityUBS PR
https://github.com/ita-social-projects/GreenCity/issues/4368

## Summary Of Changes :fire:
Modified regexp patterns on AddressDto "houseNumber" field

## Issue Link :clipboard:
_&lt;Link to the issue&gt;_

## Added
Added tests for regexp patterns

## Changed
* _&lt;detail item of what was changed&gt;_
* _&lt;describe functionality that was changed&gt;_

## Deleted
* _&lt;detail item of what was removed&gt;_

## How to test :clipboard:
It could be tested using swagger. Using UserProfileController "/user/update" endpoint (method=PUT)

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [ ] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        final Validator validator = factory.getValidator();
+
+        Set<ConstraintViolation<AddressDto>> constraintViolations =
+            validator.validate(dto);
+
+        assertThat(constraintViolations).isEmpty();
+    }
+
+    private static Stream<Arguments> provideFieldsAndValidValues() {
+        return Stream.of(
```
Comment (RomaMocherniuk) [inline service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java:57]: Case with `-` is also valid
Please test them too

File: service-api/src/main/java/greencity/constant/ValidationConstant.java [hunk 2]
```diff
@@ -9,7 +9,7 @@ public class ValidationConstant {
     public static final String CITY_EN_REGEXP = "^([A-Z][a-z]{0,39}[ʼ'`ʹ]?[a-z]{1,39}($|[ -](?=[A-Z]))){1,10}$";
     public static final String CH_EN = "[A-Za-z\\s-ʼ'`ʹ,.]";
     public static final String CH_UA = "[ЁёІіЇїҐґЄєА-Яа-я\\s-ʼ'`ʹ,.]";
-    public static final String CH_NUM = "[ЁёІіЇїҐґЄєА-Яа-яA-Za-z0-9\\s-.]";
+    public static final String CH_NUM = "[ЁёІіЇїҐґЄєА-Яа-яA-Za-z0-9-/\\s-.]";
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/constant/ValidationConstant.java]: I see you have a length limit too.
Please add test

File: service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java (hunk 1 above)
Comment (RomaMocherniuk) [reply service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java:57]: please add tests for  `ЁёІіЇїҐґЄє` too

Comment (RomaMocherniuk) [reply service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java:57]: and please add test for invalid cases

File: service-api/src/main/java/greencity/constant/ValidationConstant.java (hunk 2 above)
Comment (RomaMocherniuk) [reply service-api/src/main/java/greencity/constant/ValidationConstant.java]: why did you add `-`?
The regex already contains it

Comment (RomaMocherniuk) [reply service-api/src/main/java/greencity/constant/ValidationConstant.java:12]: What does `s` in this regex mean?
If it should be `whitespace character` then `\` should be added before
https://www.regular-expressions.info/shorthand.html
In this regex, it means `s` and could be removed

File: service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java (hunk 1 above)
Comment (RomaMocherniuk) [reply service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java:57]: add test for` whitespace characte`

File: service-api/src/main/java/greencity/constant/ValidationConstant.java (hunk 2 above)
Comment (RomaMocherniuk) [reply service-api/src/main/java/greencity/constant/ValidationConstant.java]: `@Length(max = 5)` annotation is redundant if you add `{1,5}` to the regex

File: service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java (hunk 1 above)
Comment (RomaMocherniuk) [reply service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java:57]: add test for `.` too

**Comment by RomaMocherniuk** [comment]
It seems like the PR title is incorrect
`[UBS-User. User Data page] Add an alternative email to the UBS User Data page #4368`

**Review by Markiro1 — APPROVED** [review]

**Review by juseti — APPROVED** [review]

File: service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java (hunk 1 above)
Comment (ABbondar) [reply service-api/src/test/java/greencity/dto/courier/address/AddressDtoTest.java:57]: fixed

File: service-api/src/main/java/greencity/constant/ValidationConstant.java (hunk 2 above)
Comment (ABbondar) [reply service-api/src/main/java/greencity/constant/ValidationConstant.java]: fixed

**Review by RomaMocherniuk — APPROVED** [review]

**Review by NKorzhik — APPROVED** [review]
